# Sentiment Analyses

#### Sentiment Analyses uses *Polarity* and *Subjectivity* to analyze text

## Polarity
- Measures the sentiment or emotion of a text, ranging from -1 to 1 (Negative and Positive)

## Subjectivity
- Indicates how much of the text is based on personal opinion, ranging from 0 to 1 (Objective and Subjective)

## Downloads

ite140 % pip3 install nbformat
ite140 % pip3 install spacytextblob
ite140 % pip3 install spacy
ite140 % pip3 install textblob
ite140 % pip3 install spacy download en_core_web_sm
ite140 % pip3 install https://github.com/explosion/spacy-models/releases/download/en_core_web_sm-3.5.0/en_core_web_sm-3.5.0.tar.gz


In [39]:
import spacy
from spacytextblob.spacytextblob import SpacyTextBlob


In [40]:
nlp = spacy.load('en_core_web_sm')
nlp.add_pipe('spacytextblob')

#Example 1(Edwardes)

text = 'I had a really horrible day. It was the worst day ever! But every now and then I have a really good day that makes me happy.'
doc = nlp(text)

In [41]:
doc._.blob.polarity

-0.125

A *polarity* of -0.125 indicates that the statement has an almost equal amount of positive and negative tone, but is slightly more negative

In [42]:
doc._.blob.subjectivity

0.9

A *Subjectivity* of 0.9 indicates that the text is mostly made up of words that have strong emotions

Lets do another example

In [43]:
# Example 2 (Chatgpt)
text = 'The product did not meet the advertised specifications, with defects in build quality and multiple malfunctions occuring within one week of use'
doc = nlp(text)

In [44]:
doc._.blob.polarity

0.0

A *Polarity* of-0 indicates the statement has no emotional words

In [45]:
doc._.blob.subjectivity

0.0

A *subjectivity* of 0 indicates that the statement is solely based on facts and no personal feelings or judgements whatsoever

## Summary 
- Sentiment Anaylses uses polarity and subjectivity to evaluate a statement based on its emotional words, and bias from the author

#### Soures 
- chatGPT
- Edwardes, Sam. “spacytextblob · spaCy Universe.” spaCy, https://spacy.io/universe/project/spacy-textblob. Accessed 8 October 2024.

